tests/ltp/patches/fix_sigrelse_sigrelse01.patch

```diff
@@ -0,0 +1,673 @@
+ #define DEBUG 0
+ #endif
+-
+-#define CHILD_EXIT(VAL) ((VAL >> 8) & 0377)	/* exit value of child process */
+-#define CHILD_SIG(VAL)   (VAL & 0377)	/* signal value of child proc */
+-
+-#define MAXMESG 512		/* the size of the message string */
+-
+-#define READY "ready"		/* signal to parent that child is set up */
+-
+ #define TIMEOUT 30		/* time (sec) used in the alarm calls */
+ 
+-/* child exit values */
+-#define EXIT_OK    0
+-#define SIG_CAUGHT 8
+-#define WRITE_BROK 16
+-#define HANDLE_ERR 32
+-
+ int TST_TOTAL = 1;		/* number of test items */
+ 
+ char *TCID = "sigrelse01";	/* test case identifier */
+-static char mesg[MAXMESG];	/* message buffer for tst_res */
+ static int pid;			/* process id of child */
+-static int pipe_fd[2];		/* file descriptors for pipe parent read */
+-static int pipe_fd2[2];		/* file descriptors for pipe child read */
+ static int phase;		/* flag for phase1 or phase2 of */
+ 				/* signal handler */
+ static int sig_caught;		/* flag TRUE if signal caught */
+@@ -177,6 +154,53 @@ static int sig_caught;		/* flag TRUE if signal caught */
+ /* array of counters for signals caught by handler() */
+ static int sig_array[NUMSIGS];
+ 
++void hold_signals(void)
++{
++	int sig;		/* current signal number */
++	int rv;			/* function return value */
++
++        if (setup_sigs() < 0) {
++                /* an error occured - exit from test  */
++                tst_brkm(TBROK, cleanup, "Failed to setup sugnals");
++
++        } else {
++                /* all set up to catch signals, now hold them */
++
++                for (sig = 1; sig < NUMSIGS; sig++) {
++                        if (choose_sig(sig)) {
++                                if ((rv = sighold(sig)) != 0) {
++                                        /* THEY say sighold ALWAYS returns 0 */
++                                        tst_brkm(TBROK|TERRNO, cleanup, "Failed to do sighold, returned: %d", rv);
++                                }
++                        }
++                }
++
++        }
++
++}
++
++void release_signals(void)
++{
++	int sig;		/* current signal number */
++	int rv;			/* function return value */
++	for (sig = 1; sig < NUMSIGS; sig++) {
++        	if (choose_sig(sig)) {
++			/* all set up, release and catch a signal */
++
++			sig_caught = FALSE;     /* handler sets it to TRUE when caught */
++			if ((rv = sigrelse(sig)) != 0) {
++				/* THEY say sigrelse ALWAYS returns 0 */
++				tst_brkm(TBROK|TERRNO, cleanup, "sigrelse did not return 0. rv:%d", rv);
++			}
++
++			/* give signal handler some time to process signal */
++			wait_a_while();
++        	}
++
++        }                       /* endfor */
++
++}
++
+ /***********************************************************************
+  *   M A I N
+  ***********************************************************************/
+@@ -192,9 +216,6 @@ int main(int argc, char **argv)
+ 	 * parse standard options
+ 	 */
+ 	tst_parse_opts(argc, argv, NULL, NULL);
+-#ifdef UCLINUX
```

Comment:
Why is this removed?  We should be using the UCLINUX code paths because we are a uCLinux implementation.

Reply:
uClinux code is removed because exec is the problem and I will update the same in the patch description.

Reply:
I updated the patch description.